app/controllers/users/checkins_controller.rb

```diff
@@ -27,6 +27,16 @@ def create
     flash[:notice] = 'Checked in.'
   end
 
+  def import
+    if params[:file] && valid_file?
```

Comment:
I'd suggest to move this logic into separate class. But not a model class.
You guys have [services](https://github.com/earlymarket/Coposition/tree/master/app/services) already. So it can be new service like `ImportCheckins`.

Going further, I would also suggest to add something like [interactor](https://github.com/collectiveidea/interactor/) to make single interface for all service classes and to make it easier to get result (status) from service calls.